libfreerdp/core/nla.c

```diff
@@ -1552,63 +1580,80 @@ int nla_recv_pdu(rdpNla* nla, wStream* s)
 	WINPR_ASSERT(nla);
 	WINPR_ASSERT(s);
 
-	if (nla_decode_ts_request(nla, s) < 1)
-		return -1;
-
-	if (nla->errorCode)
+	if (nla_get_state(nla) == NLA_STATE_EARLY_USER_AUTH)
 	{
 		UINT32 code;
+		Stream_Read_UINT32(s, code);
+		if (code != 0x00000000)
+		{
+			WLog_DBG(TAG, "Early User Auth active: FAILURE");
```

Comment:
maybe log the result code too (should only be 2 allowed, but you never know when that will change)

Reply:
I agree with you, code updated